File: coding/05th July 2025/Find Lucky Integer in an Array.java
//Find Lucky Integer in an Array
class Solution 
{
    public int findLucky(int[] arr) 
    {
        int result = -1;
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int ele : arr)
        {
            if(map.containsKey(ele))
                map.put(ele, map.get(ele) + 1);
            else
                map.put(ele, 1);
        }
        for(int key : map.keySet())
        {
            if(key == map.get(key) && key > result)
                result = key;
        }
        return result;
    }
}

File: Coding in Python/10th February 2026/Find Lucky Integer in an Array.py
#Find Lucky Integer in an Array
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        dict1 = {}
        result = -1
        for ele in arr:
            if ele in dict1:
                dict1[ele] += 1
            else:
                dict1[ele] = 1
        for k in dict1:
            if k == dict1[k] and k > result:
                result = k
        return result
